BUILD.md

```diff
@@ -37,6 +37,7 @@ As a convenience, a `build.sh` script is provided which can be used to execute t
 $ ./build.sh                           # build the cuML libraries, tests, and python package, then
                                        # install them to $INSTALL_PREFIX if set, otherwise $CONDA_PREFIX
 ```
+For workflows that involve frequent switching among branches or between debug and release builds, it is recommended that you install [ccache](https://ccache.dev/) and make use of it by passing the `--ccache` flag to `build.sh`.
```

Comment:
I think adding the instructions for directly invoking cmake would be a good idea for people that don't go through the `build.sh` script.